src/rhsmlib/dbus/objects/register.py

```diff
@@ -104,6 +104,12 @@ def __init__(self, conn=None, object_path=None, bus_name=None):
     def Start(self, locale, sender=None):
         locale = dbus_utils.dbus_to_python(locale, expected_type=str)
         Locale.set(locale)
+
+        # When somebody else started domain socket listener, then
+        if self.impl.server is not None:
+            log.warning(f"domain socket listener already running, opened by: {self.impl.server.sender}")
```

Comment:
might be better to include `_caller_uid` here. it's probably a little more recognizable in a log than a d-bus unique name.

Reply:
also, just to summarize a slack conversation between @jirihnidek @owtaylor and me. This causes a few other problems:

1. There's still the DoS problem because one user can call `Start()` and prevent another user from calling `Start()`
2. If the user session crashes during login before `Stop()` is called, they'll have to reboot to register instead of just log back in
3. gnome-settings-daemon currently has unpaired `Start()` in its unregister path. So if user unregisters they'll have to reboot to register again

So I think `Start()` needs to silently succeed if it's already started to maintain backward compatibility.  The two easiest ways to do that are:

1. go back to tracking a set of caller uids like the sketch from https://github.com/candlepin/subscription-manager/pull/3350#issuecomment-1787695472
2.  Allow `Start()` from multiple users but make `Stop()` a no-op unless it's called by root (as mentioned earlier in https://github.com/candlepin/subscription-manager/pull/3350#issuecomment-1787643938 )